### PR TITLE
Query and update animals table

### PR DESCRIPTION
**Pull request by tobuya**
- Add a column `species` of type string to the animals table.
- Insert the following data into the animal table
Screenshot 2023-03-01 233530

-  Update the animals table by setting the `species` column to `unspecified`. Verify the change was made. Then roll back the change and verify that the `species` column went back to the state before the transaction.
Screenshot 2023-03-01 234958
Screenshot 2023-03-01 235111

- Update animals table by setting the `species` column to `digimon`  for animals that have a name ending in `mon` and `pokemon` for all animals that don't have `species` already set.
Screenshot 2023-03-01 235308

-  Inside a transaction delete all records in the animals table, then roll back the transaction. After the rollback verify if all records in the animals table still exist.
Screenshot 2023-03-01 235724
Screenshot 2023-03-01 235826

- Delete all animals born after Jan 1st, 2022
- Create a savepoint for the transaction.
- Create a savepoint for the transaction.
- Rollback to the savepoint
- Update all animals' weights that are negative to be their weight multiplied by -1
- Commit transaction
Screenshot 2023-03-02 000331
Screenshot 2023-03-02 000446

- Count the number of animals
Screenshot 2023-03-02 001159

- Count animals that have never tried to escape
Screenshot 2023-03-02 001257

- Average weight of animals
Screenshot 2023-03-02 001413

- Find who escapes the most, neutered or not neutered animals
Screenshot 2023-03-02 001556

- Find the minimum and maximum weight of each type of animal
Screenshot 2023-03-02 001742

- Find  the average number of escape attempts per animal type of those born between 1990 and 2000
Screenshot 2023-03-02 001922

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -1,8 +1,18 @@
...
 ('Gabumon', 'Nov 15, 2018', 8, 'true', 2),
 ('Pikachu', 'Jan 7, 2021', 15.04, 'false', 1),
 ('Devimon', 'May 12, 2017', 11, 'true', 5);
+
+INSERT INTO animals(name, date_of_birth, weight_kg, neutered, escape_attempts)
+VALUES
+('Charmander', 'Feb 8, 2020', 11,'false', 0 ),
+('Plantmon', 'Nov 15, 2021', 5.7,'true', 2 ),
+('Squirtle', 'Apr 2, 1993', 12.13, 'false', 3 ),
+('Angemon',  'Jun 12, 2005', 45,  'true', 1),
```
Comment (omar-muhamad) [inline data.sql]: - [ ] The weight of these 4 animals suppose to be a negative value, Please fix that.

image

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @tobuya,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: data.sql (hunk 1 above)
Comment (tobuya) [reply data.sql]: @Omar-Muhamad Thank you so much for the observation, am on it.

**Review by Alfredbis29 — APPROVED** [review]
Hi @tobuya , 

**### APPROVED**
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
You documented your work professionally. :white_check_mark: 
You used the correct workflow. :white_check_mark: 
Everything that creates or change the tables structure are saved in the schema.sql file. :white_check_mark: 

## Optional suggestions

- [ ] I kindly request that you take the necessary steps to correct the license in your readme file and ensure that it is properly adhered to. This will not only protect your rights as the creator of the software, but it will also help maintain a healthy open source community. :wink: , kindly fix this :wink: 
Screenshot from 2023-03-01 23-44-11


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments @Alfredbis29 in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**